src/cmd/tools/m3ctl/main/main.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/m3db/m3/src/cmd/tools/m3ctl/apply"
+	"github.com/m3db/m3/src/cmd/tools/m3ctl/namespaces"
+	"github.com/m3db/m3/src/cmd/tools/m3ctl/placements"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+const (
+	DefaultEndpoint = "http://localhost:7201"
+)
+
+var (
+	endPoint string
+	yamlPath string
+	showAll  bool
+	nodeName string
+)
+
+func main() {
+
+	zapper, err := zap.NewDevelopment()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+
+	rootCmd := &cobra.Command{
+		Use: "cobra",
+	}
+
+	getCmd := &cobra.Command{
+		Use:   "get",
+		Short: "Get specified resources from the remote",
+	}
+
+	deleteCmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete specified resources from the remote",
+	}
+
+	applyCmd := &cobra.Command{
+		Use:   "apply",
+		Short: "Apply various yamls to remote endpoint",
+		Long:  `This will take specific yamls and send them over to the remote
+endpoint.  See the yaml/examples directory for examples.  Operations such as 
+database creation, database init, adding a node, and replacing a node, are supported.
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+
+			zapper.Debug(fmt.Sprintf("Running command:%s:\n", cmd.Name()))
+
+			zapper.Debug(fmt.Sprintf("args:%v:\n", cmd.LocalFlags().Lookup("file").Value.String()))
```

Comment:
Best to use non sugared logging here, using fields. Also as per effective go all log statements and error messages should start with non-capital letters.
```
logger.Debug("command arguments", zap.String("args", cmd.LocalFlags().Lookup("file").Value.String()))
```

Effective Go:
https://golang.org/doc/effective_go.htm